apps/settings/js/panels/apn_settings/panel.js

```diff
@@ -9,38 +9,25 @@ define(function(require) {
   var SettingsPanel = require('modules/settings_panel');
   var ApnSettingsManager = require('modules/apn/apn_settings_manager');
   var Toaster = require('shared/toaster');
+  var DialogService = require('modules/dialog_service');
 
   return function ctor_apn_settings_panel() {
     var _serviceId = 0;
 
     var _rootElement;
-    var _warningDialog;
-    var _warningDialogOkBtn;
-    var _warningDialogCancelBtn;
 
     var _showResetApnWarningDialog = function(callback) {
```

Comment:
Because the callback should only functional with `submit` action, we can remove this method and show the dialog in `_resetApn`,